sidechain/consensus/slots/src/lib.rs

```diff
@@ -199,9 +210,23 @@ pub trait SimpleSlotWorker<B: ParentchainBlock> {
 			);
 		}
 
-		let _ = self.claim_slot(&slot_info.parentchain_head, slot, &epoch_data)?;
-
-		let proposer = match self.proposer(slot_info.parentchain_head.clone(), shard) {
+		let _claim =
+			self.claim_slot(&slot_info.last_imported_parentchain_head, slot, &epoch_data)?;
+
+		// Import and retrieve the parentchain header on which the new sidechain block is based on.
+		let latest_imported_parentchain_header =
+			match self.import_latest_parentchain_block(&slot_info.last_imported_parentchain_head) {
+				Ok(p) => p,
+				Err(e) => {
+					warn!(
+						target: logging_target,
+						"Failed to import and retrieve latest parentchain block header: {:?}", e
+					);
+					return None
+				},
+			};
```

Comment:
Trigger the import of the latest parentchain block once we have claimed a slot.

Reply:
Clean enhancement!